test/runnable/interface2.d

```diff
@@ -920,6 +923,57 @@ void test27()
 }
 
 /*******************************************************/
+// 1747 & 2013
+
+void test1747()
+{
+    interface IA          { int mA(); }
+    interface IB : IA     { int mB(); }
+    interface IC : IB     { }
+    interface ID : IA, IC { int mD(); }
+
+    // offset:   0   +n  +n + ptrsize
+    //                  (IA)
+    //                   IB
+    //               IA, IC
+    static class C : ID
+    {
+        int mA() { return 1; }
+        int mB() { return 2; }
+        int mD() { return 3; }
+    }
+
+    C c = new C;    void* pc  = *cast(void**)&c;
+    ID id = c;      void* pid = *cast(void**)&id;
+    IC ic = c;      void* pic = *cast(void**)&ic;
+    IB ib = c;      void* pib = *cast(void**)&ib;
+    IA ia = c;      void* pia = *cast(void**)&ia;
+
+    //printf(" c = %p\n", pc);
+    //printf("id = %p\n", pid);
+    //printf("ic = %p\n", pic);
+    //printf("ib = %p\n", pib);
+    //printf("ia = %p\n", pia);
+
+    size_t n = pid - pc;
+    assert(pic == pc + n + (void*).sizeof);
+    assert(pib == pc + n + (void*).sizeof);     // OK <- NG
+    assert(pia == pc + n);
```

Comment:
Is the memory layout fixed by the ABI? If not, these asserts might be DMD specific.